### PR TITLE
formatNumberReadable now prints signed integers as well

### PR DESCRIPTION
**Pull request by Ruko97**
Fixes issue #9601

### CODE REVIEW

File: libsolutil/StringUtils.h [hunk 1]
```diff
@@ -101,7 +101,7 @@ std::string joinHumanReadablePrefixed
 		return _separator + joinHumanReadable(_list, _separator, _lastSeparator);
 }
 
-/// Formats large numbers to be easily readable by humans.
+/// Formats unsigned large numbers to be easily readable by humans.
```
Comment (Marenz) [inline libsolutil/StringUtils.h]: I think `Formats large unsigned numbers` reads nicer

File: libsolutil/StringUtils.h [hunk 2]
```diff
@@ -111,7 +111,7 @@ std::string joinHumanReadablePrefixed
 /// like  0x5555...{+56 more}...5555
 /// @example formatNumber((u256)0x7ffffff)
```
Comment (Marenz) [inline libsolutil/StringUtils.h]: I would suggest to keep only one documentation and just document what's different in the other function, e.g. here you could just say `/// Same as @ref formatNumberReadable but only for unsigned numbers`

File: libsolutil/StringUtils.h [hunk 3]
```diff
@@ -181,6 +181,35 @@ inline std::string formatNumberReadable(
...
 
+/// Formats large numbers to be easily readable by humans.
+/// Returns decimal representation for smaller numbers; hex for large numbers.
+/// "Special" numbers, powers-of-two and powers-of-two minus 1, are returned in
+/// formulaic form like 0x01 * 2**24 - 1.
+/// @a T can be any integer variable, will typically be u160, u256 or bigint.
+/// @param _value to be formatted
+/// @param _useTruncation if true, internal truncation is also applied,
+/// like  0x5555...{+56 more}...5555
+/// @example formatNumber((u256)0x7ffffff)
```
Comment (Marenz) [inline libsolutil/StringUtils.h]: Example is still from a very old version, better update it :)

**Review by Marenz — CHANGES_REQUESTED** [review]

**Comment by Marenz** [comment]
Looks good. The end result is certainly more readable, though as @cameel [mentioned](https://github.com/ethereum/solidity/issues/9601#issuecomment-739534814) getting rid of the `0x80` part would be best. Doesn't have to happen in this PR though, this is still better than what we had before.

**Comment by Ruko97** [comment]
I modified the comments according to your advice. I'll get rid of the 0x80 part in another PR.

**Comment by Ruko97** [comment]
Also please ignore my "dismissed a stale review" and my temporary closing of this PR. This is my first PR, so i am not fully familiar with the interface

File: libsolutil/StringUtils.h [hunk 4]
```diff
@@ -194,12 +194,10 @@ inline std::string formatNumberReadable(
...
 	);
 
-	if (_value >= 0)
-	{
+	if (_value >= 0) {
 		bigint const _v = bigint(_value);
 		return formatUnsignedNumberReadable(_v, _useTruncation);
-	} else
-	{
+	} else {
```
Comment (Marenz) [inline libsolutil/StringUtils.h]: The previous style was alright :)

Comment (Marenz) [reply libsolutil/StringUtils.h]: That's also the reason the coding style check failed

Comment (Ruko97) [reply libsolutil/StringUtils.h]: Sorry, I accidentally changed the file's style :(

Comment (Ruko97) [reply libsolutil/StringUtils.h]: Fixed

**Comment by Ruko97** [comment]
@Marenz  Shall I resend this pull request? This one has become cluttered with the updating coding style commits

**Comment by Marenz** [comment]
No no, just amend (as in, merge into one) the commits and force push, then it's all clean again

**Comment by Ruko97** [comment]
@Marenz  t_native_text_ext_chainlink fails, but I only changed the comments here. Should I try and fix it?

**Comment by Marenz** [comment]
> @Marenz t_native_text_ext_chainlink fails, but I only changed the comments here. Should I try and fix it?

I don't think it was your change. It's not marked as "required" as well, so you can just ignore it for now

**Comment by Marenz** [comment]
We still have three commits that should be merged into one though :)-

**Comment by Ruko97** [comment]
ok, I'll merge the 3 commits together

File: libsolutil/StringUtils.h [hunk 5]
```diff
@@ -181,6 +173,38 @@ inline std::string formatNumberReadable(
...
+	static_assert(
+		std::numeric_limits<T>::is_integer,
+		"only integer numbers are supported"
+	);
+
+	if (_value >= 0)
+	{
+		bigint const _v = bigint(_value);
+		return formatUnsignedNumberReadable(_v, _useTruncation);
+	} else
```
Comment (Marenz) [inline libsolutil/StringUtils.h]: Could you change this so that the `else` is on its own line wtihout the `}`

**Review by Marenz — CHANGES_REQUESTED** [review]
Looking good, just one minor change

File: libsolutil/StringUtils.h (hunk 5 above)
Comment (Ruko97) [reply libsolutil/StringUtils.h]: Sure :D

**Review by Marenz — APPROVED** [review]

**Pull request merged**